### PR TITLE
RFC implementation

### PR DESCRIPTION
**Pull request by Vitaly-Protasov**
Solved#436

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#472](https://github.com/yegor256/aibolit/issues/472) is assigned to @aravij[/z](https://www.0crat.com/u/aravij), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/bb5e9d64-990b-453e-b626-b0b5a0fa2c17); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@692aa6ed-a9f8-4a33-8b6d-3271975232f3, thread: PQ-CT5UB7XAP
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 2 error(s) in XML document: -1:-1: cvc-pattern-valid: Value 'gh:Aibolit (multi-target)' is not facet-valid with respect to pattern '[a-z]{2}:[A-Z0-9a-z.\-#/]+' for type 'milestoneID'.;-1:-1: cvc-complex-type.2.2: Element 'successor' must have no element [children], and the value must be valid.
```

[0.53.15](https://github.com/zerocracy/farm): CID: [18d0363d-a815-4550-842c-f6dd6c519097](https://www.0crat.com/footprint/CT5UB7XAP/18d0363d-a815-4550-842c-f6dd6c519097), Type: "Job milestoned"

File: aibolit/metrics/RFC/rfc.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        initialized_method_vars = set()
+
+        p = JavaPackage(filename)
+        for class_name in p.java_classes:
+            tree = p.java_classes[class_name]
+            for class_method in tree.subtrees_with_root_type(ASTNodeType.METHOD_DECLARATION):
+                ast_each_method = AST(tree.tree.subgraph(class_method), class_method[0])
+                # to form a set of all methods in the class
+                names = list(ast_each_method.children_with_type(ast_each_method.root, ASTNodeType.STRING))
+                method_name = tree.get_attr(names[0], 'string')
```
Comment (acheshkov) [inline aibolit/metrics/RFC/rfc.py:26]: > tree.get_attr(names[0], 'string')
@aravij  It's not usual that to get the name of the method we have to call the method of the some AST tree

File: aibolit/metrics/RFC/rfc.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                        current_name = name_of_invoked_class.method_name
+                        if not current_name.startswith('/'):
+                            if current_name not in initialized_method_vars:
+                                called_methods.add(current_name)
+
+                # to count invoked methods without initialization the results into new variable
+                for inv_method in ast_each_method.nodes_by_type(ASTNodeType.METHOD_INVOCATION):
+                    name_of_invoked_class = ast_each_method.get_method_invocation_params(inv_method)
+                    current_name = name_of_invoked_class.method_name
+                    if current_name not in initialized_method_vars:
```
Comment (acheshkov) [inline aibolit/metrics/RFC/rfc.py:48]: @Vitaly-Protasov should we check element existence if we add to set?

File: aibolit/metrics/RFC/rfc.py (hunk 1 above)
Comment (Vitaly-Protasov) [reply aibolit/metrics/RFC/rfc.py]: rewrrited

**Comment by acheshkov** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job audit: performer @aravij[/z](https://www.0crat.com/u/aravij)[/z](https://www.0crat.com/u/aravij) didn't make CR comments

**Pull request merged**